src/community/china.md

```diff
@@ -6,74 +6,171 @@ toc: true
+   guide and continue from that procedure.
+
+From this example, `flutter pub get` fetches packages from `flutter-io.cn`.
+in any terminal where you set `PUB_HOSTED_URL` and `FLUTTER_STORAGE_BASE_URL`.
+
+Any environment variables set using `export` in this procedure
+only apply to the current terminal window.
+To set these values on a permanent basis,
+add those three `export` commands to the bottom of the `*rc` or `*_profile`
+file that your preferred shell uses.
+
+### Download other versions of Flutter
+
+To install a different version of Flutter found in the
+[archive]({{site.url}}/release/archive),
+replace `storage.googleapis.com` with that of your trusted mirror.
+This should improve download speed.
+
+For example:
 
 * Original URL:<br>
   [`https://storage.googleapis.com/{{path}}`](https://storage.googleapis.com/{{path}})
 
 * Mirrored URL:<br>
   [`https://storage.flutter-io.cn/{{path}}`](https://storage.flutter-io.cn/{{path}})
 
-You must also set two environment variables to upgrade Flutter and use the pub
-package repository in China. Instructions are below.
+## Configure your machine to publish your package
 
-{{site.alert.important}}
-  Use mirror sites only if you _trust_ the provider.
-  The Flutter team cannot verify their reliability or security.
-{{site.alert.end}}
+If you receive errors when attempting to publish your package,
+try one of the following solutions:
 
-## Configuring Flutter to use a mirror site
-
-If you're installing or using Flutter in China,
-it might be helpful to use a trustworthy local
-mirror site that hosts Flutter's dependencies.
-To instruct the Flutter tool to use an alternate storage location,
-you need to set two environment variables, `PUB_HOSTED_URL` and
-`FLUTTER_STORAGE_BASE_URL`, before running the `flutter` command.
-
-Taking macOS or Linux as an example, here are the first few steps in
-the setup process for using a mirror site. Run the following in a Bash
-shell from the directory where you wish to store your local Flutter clone:
-
-```terminal
-$ export PUB_HOSTED_URL=https://pub.flutter-io.cn
-$ export FLUTTER_STORAGE_BASE_URL=https://storage.flutter-io.cn
-$ git clone -b dev {{site.repo.flutter}}.git
-$ export PATH="$PWD/flutter/bin:$PATH"
-$ cd ./flutter
-$ flutter doctor
-```
+  {% comment %}
+  From https://github.com/flutter/website/pull/9338#discussion_r1328077020
+  {% endcomment %}
+
+1. Clear your `PUB_HOSTED_URL` environment variable.
+
+   ```terminal
+   $ unset PUB_HOSTED_URL
+   ```
+
+   Remember to reset this environment variable after you publish.
+
+   ```terminal
+   export PUB_HOSTED_URL=https://pub.flutter-io.cn
+   ```
+
+1. Configure a proxy.
+   To configure a proxy, check out the [Dart documentation on proxies][].
+
+  {% comment %}
+  From https://github.com/flutter/website/issues/2556#issuecomment-481566476
+  {% endcomment %}
+
+1. Enable your VPN without adding a `credentials.json` file.
+
+1. Disable your proxy and add a `credentials.json` file.
```

Comment:
**question:** @Realank, @chenglu: Does this look correct? Do you have terminal buffer that shows this working that you can share? I'd like to be able to show the successful result of this change.

Reply:
(@Realank is not available for the answer probably) If we've unset the env before, we shall no longer have the *Enable* and *Disable* steps. Configure a proxy and you should be good to go.